specs/networking/libp2p-standardization.md

```diff
@@ -0,0 +1,192 @@
+ETH 2.0 Networking Spec - Libp2p standard protocols
+===
+
+# Abstract
+
+Ethereum 2.0 clients plan to use the libp2p protocol networking stack for
+mainnet release. This document aims to standardize the libp2p client protocols,
+configuration and messaging formats.
+
+# Libp2p Protocols
+
+## Transport
+
+This section details the libp2p transport layer that underlies the
+[protocols](#protocols) that are listed in this document.
+
+Libp2p allows composition of multiple transports. Eth2.0 clients should support
+TCP/IP and optionally websockets. Websockets are useful for implementations
+running in the browser and therefore native clients would ideally support these implementations 
+by supporting websockets.
+
+An ideal libp2p transport would therefore be TCP/IP with a fallback to
+websockets.
+
+### Encryption
+
+Libp2p currently offers [Secio](https://github.com/libp2p/specs/pull/106) which
+can upgrade a transport which will then encrypt all future communication. Secio
+generates a symmetric ephemeral key which peers use to encrypt their
+communication. It can support a range of ciphers and currently supports key
+derivation for elliptic curve-based public keys.
+
+Current defaults are:
+- Key agreement: `ECDH-P256` (also supports `ECDH-P384`)
+- Cipher: `AES-128` (also supports `AES-256`, `TwofishCTR`)
+- Digests: `SHA256` (also supports `SHA512`)
+
+
+## Protocols
+
+This section lists the necessary libp2p protocols required by Ethereum 2.0
+running a libp2p network stack.
+
+## Multistream-select
+
+#### Protocol id: `/multistream/1.0.0`
+
+Clients running libp2p should support the [multistream-select](https://github.com/multiformats/multistream-select/)
```

Comment:
I'll be writing a primer about connection bootstrapping and protocol negotiation soon; it's confusing.

Reply:
Awesome. Let's link to it here :)